erpnext/accounts/doctype/pos_profile/pos_profile.js

```diff
@@ -68,6 +64,10 @@ frappe.ui.form.on('POS Profile', {
 				}
 			};
 		});
+
+		erpnext.queries.setup_queries(frm, "Warehouse", function() {
+			return erpnext.queries.warehouse(frm.doc);
+		});
```

Comment:
Merge both the function

Reply:
@hrwX  Not understood  what are you trying to say

Reply:
put the `onload` inside `frappe.ui.form.on`

Reply:
Refactored onload in pos profile